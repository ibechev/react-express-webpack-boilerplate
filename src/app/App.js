import React from "react";
import PropTypes from "prop-types";

const App = ({ message }) => {
  return (
    <div className="app">
      <h1>{message}</h1>
      <ul>
        <li>Express</li>
        <li>React</li>
        <li>Webpack</li>
        <li>HMR</li>
        <li>SCSS</li>
        <li>Jest</li>
        <li>Enzyme</li>
        <li>ESLint</li>
      </ul>
    </div>
  );
};

App.propTypes = {
  message: PropTypes.string.isRequired
};

export default App;
