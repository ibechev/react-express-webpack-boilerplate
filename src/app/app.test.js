import React from "react";
import renderer from "react-test-renderer";
import { propTypeErrors } from "../../tests/config/testUtils";
import App from "./App";

describe("Expect <App />", () => {
  const defaultProps = {
    message: "Hey hey hey"
  };

  it("to render correctly", () => {
    const wrapper = renderer.create(<App {...defaultProps} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("to not throw error with correct props", () => {
    propTypeErrors(<App {...defaultProps} />, defaultProps);
  });
});
