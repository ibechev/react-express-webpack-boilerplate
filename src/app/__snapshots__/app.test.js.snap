// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <App /> to render correctly 1`] = `
<div
  className="app"
>
  <h1>
    Hey hey hey
  </h1>
  <ul>
    <li>
      Express
    </li>
    <li>
      React
    </li>
    <li>
      Webpack
    </li>
    <li>
      HMR
    </li>
    <li>
      SCSS
    </li>
    <li>
      Jest
    </li>
    <li>
      Enzyme
    </li>
    <li>
      ESLint
    </li>
  </ul>
</div>
`;
